aten/src/ATen/native/cpu/HistogramKernel.cpp

```diff
@@ -0,0 +1,209 @@
+#include <ATen/native/Histogram.h>
+
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/Parallel.h>
+#include <ATen/native/BucketizationUtils.h>
+
+#include <algorithm>
+#include <mutex>
+#include <tuple>
+
+namespace at { namespace native {
+
+namespace {
+
+const int64_t HISTOGRAM_GRAIN_SIZE = 200;
+
+template<typename input_t, typename output_t, typename ExtractWeight_t, bool LinearBinEdges>
+void histogram_cpu_contiguous(Tensor& hist, const Tensor& bin_edges, const Tensor& input, const Scalar& min, const Scalar& max, ExtractWeight_t extract_weight) {
+    int64_t numel_in = input.numel();
+    if (!numel_in) {
+        return;
+    }
+
+    int64_t numel_be = bin_edges.numel();
+    TORCH_CHECK(numel_be > 0, "expected bin_edges to contain at least 1 element but it contains 0");
```

Comment:
This check seems OK for the more general error-checking function?

Reply:
This is already tested in pre_check, so it's just unnecessary